runtime/common/src/apis.rs

```diff
@@ -262,18 +262,55 @@ macro_rules! impl_runtime_apis_plus_common {
 					};
 					let is_transactional = false;
 					let validate = true;
+
+					let mut estimated_transaction_len = data.len() +
+						// to: 20
+						// from: 20
+						// value: 32
+						// gas_limit: 32
+						// nonce: 32
+						// 1 byte transaction action variant
+						// chain id 8 bytes
+						// 65 bytes signature
+						210;
+					if max_fee_per_gas.is_some() {
+						estimated_transaction_len += 32;
+					}
+					if max_priority_fee_per_gas.is_some() {
+						estimated_transaction_len += 32;
+					}
+					if access_list.is_some() {
+						estimated_transaction_len += access_list.encoded_size();
+					}
+
+					let gas_limit = gas_limit.min(u64::MAX.into()).low_u64();
+					let without_base_extrinsic_weight = true;
+
+					let (weight_limit, proof_size_base_cost) =
+						match <Runtime as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+							gas_limit,
+							without_base_extrinsic_weight
+						) {
+							weight_limit if weight_limit.proof_size() > 0 => {
+								(Some(weight_limit), Some(estimated_transaction_len as u64))
+							}
+							_ => (None, None),
```

Comment:
Why `(None, None)` if there's no proof size in the `GasWeightMapping`?

`GasWeightMapping::gas_to_weight` can't really do anything useful with proof_size, can it? It only knows some unit of gas, which has no notion of different resource types. And unless I'm missing something, it returns 0 unconditionally in our case.